cms/models.py

```diff
@@ -69,3 +76,33 @@ def get_context(self, request, *args, **kwargs):
         context = super(ResourcePage, self).get_context(request)
 
         return context
+
+class BootCampRunPage(BootcampProductPage):
+    """
+    CMS page representing a klass
+    """
+
+    template = "product_page.html"
+
+    klasses = models.OneToOneField(
+        "klasses.Klass",
+        null=True,
+        on_delete=models.SET_NULL,
+        help_text="The Klass for this page",
+    )
+
+    content_panels = BootcampProductPage.content_panels + [FieldPanel("klasses")]
+    
+    def get_context(self, request, *args, **kwargs):
+        """
+        return page context.
+        """
+        context = super(BootCampRunPage, self).get_context(request)
+        return context
+
+    def save(self, *args, **kwargs):
+        # autogenerate a unique slug so we don't hit a ValidationError
+        if not self.title:
+            self.title = self.__class__._meta.verbose_name.title()
+        self.slug = slugify("{}-{}".format(self.title, self.id))
```

Comment:
Given that these URLs are going to be shared I think we should have a better slug pattern than `{title}-{id}`. Plus this looks like it was taken from xPRO `CourseProgramChildPage` which isn't intended to be used standalone that's why this slugging pattern works there but might need tweaks as these will be indexed by search engines as well.

Reply:
suggestion  ?